lib/src/main/java/com/knuddels/jtokkit/TokenEncoder.java

```diff
@@ -101,13 +104,32 @@ public int addTokensAndGetCount(int maxTokenCount, boolean keepEncodings, byte[]
             return 1;
         } else {
             int length = match.length();
-            return addTokensAndGetCount(maxTokenCount, keepEncodings, out, ranks, match, length);
+            if (length < VERY_LARGE_TOKENIZER_BYTE_THRESHOLD) {
+                return addTokensAndGetCountSmall(maxTokenCount, keepEncodings, out, ranks, match, length);
+            } else {
+                return addTokensAndGetCountLarge(this, maxTokenCount, keepEncodings, out, match, length);
```

Comment:
this is the linear merge, used for very big fragments, such as "aaaaaa...aaaaaa" - the original one does it in an exponential-like manner, this one is basically linear to the number of characters